### ISSUE TITLE
Allow isomorphic apps to mount to existing DOM element

### ISSUE THREAD

**Issue by ebenpack** [dmjio/miso#455]
**Is your feature request related to a problem? Please describe.**
`startApp` allows mounting to an existing DOM node using `mountPoint`, but currently there seem to be a few impediments to isomorphic apps doing likewise. Primarily, the `miso` function doesn't take `mountPoint` into account, and `copyDOMIntoVTree` starts walking the DOM from body. From what I can tell this forces any isomorphic app to have to take control of the entire DOM from the body down.

My personal use-case is just to have some presentational markup wrapping the application. If the app could be mounted to an existing DOM element, this markup could be moved out to the initial server-side rendering. Whereas without this capability, the presentational markup needs to be pushed into the miso app, where it potentially needs to be maintained throughout the application lifecycle. I found that this seemed to require a lot of duplication of this markup (or composition of views), at the top level of every 'page', and it felt like it added some unnecessary complexity to the application.

Admittedly for this use-case such a change would mostly be a somewhat minor quality-of-life type improvement, but I can imagine this could be beneficial for other reasons as well.

I think this would also bring `startApp` and `miso` a bit closer together conceptually. I know I didn't find it clear from the documentation the way `mountPoint` worked, and I assumed I could use it with an isomorphic app, and I was fairly confused about why this wasn't working for me until I started to dig into the code.

**Describe the solution you'd like**
The solution I would propose (and I have a PR ready if you're amenable to it) is to read `mountPoint` in the `miso` function, pass it (if there is one) into `copyDOMIntoVTree`, and start walking down the DOM from that point. This will not functionally change any applications that are not specifying a mountpoint, as the mountpoint in this case will be the body, which is what's currently hard-coded into `copyDOMIntoVTree`. 

The one major risk with this change would be that it could cause isomorphic applications that are also defining a mountpoint to break. This doesn't seem to be a common use-case, though, and I suspect any such applications would already have other problems (namely that I suspect event delegation would have some major issues).

**Describe alternatives you've considered**
N/A

**Additional context**
https://github.com/ebenpack/miso/commit/63551fd06ad9b6088b39ba15692684a53e20534d

**Comment by ebenpack** [comment]
🤦 I found another way to do what I was trying to do that's already supported. That said, I still think there could be some value to this change, but I'd be interested in hearing other's thoughts.

**Comment by ebenpack** [comment]
Basically I just found the highest spot in the VDOM where I could lift this static, top-level presentational markup (I must not have been looking very hard before). See [here](https://github.com/ebenpack/planning-euchre/blob/master/common/src/Common/View.hs#L28). 

So this markup is still controlled by miso (and it still gets rendered in the initial SSR), but now at least I don't need to worry about having to duplicate it across multiple views. This is built on [miso-isomorphic-stack](https://github.com/Tehnix/miso-isomorphic-stack), which uses servant. I don't know how common this sort of approach would be for an isomorphic miso app, but if I had to guess it seems like it would be a pretty standard pattern to have a top-level routing component like this.

So, yeah, this approach works well enough for what I was trying to do. I still feel like allowing isomorphic apps to define a mount point could be a (perhaps only marginally) beneficial change, though. It was less than 10 lines to implement this, so if you're interested I'll open a PR, and it should be pretty quick to review.

Anyway, I'm really loving this project, so thanks for all your hard work!
